src/main/java/seedu/algobase/ui/ListElementPointer.java

```diff
@@ -0,0 +1,114 @@
+package seedu.algobase.ui;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+
+//@@author le0tan-reused
+//Reused from addresbook-level-4
+/**
+ * Has a cursor that points to an element in the list, and is able to iterate through the list.
+ * This is different from {@code ListIterator}, which has a cursor that points in between elements.
+ * The {@code ListIterator}'s behaviour: when making alternating calls of {@code next()} and
+ * {@code previous()}, the same element is returned on both calls.
+ * In contrast, {@code ListElementPointer}'s behaviour: when making alternating calls of
+ * {@code next()} and {@code previous()}, the next and previous elements are returned respectively.
+ */
+public class ListElementPointer {
+    private List<String> list;
+    private int index;
+
+    /**
+     * Constructs {@code ListElementPointer} which is backed by a defensive copy of {@code list}.
+     * The cursor points to the last element in {@code list}.
+     */
+    public ListElementPointer(List<String> list) {
+        this.list = new ArrayList<>(list);
+        index = this.list.size() - 1;
+    }
+
+    /**
+     * Appends {@code element} to the end of the list.
+     */
+    public void add(String element) {
+        list.add(element);
+    }
+
+    /**
+     * Returns true if calling {@code #next()} does not throw a {@code NoSuchElementException}.
+     */
+    public boolean hasNext() {
+        int nextIndex = index + 1;
+        return isWithinBounds(nextIndex);
+    }
+
+    /**
+     * Returns true if calling {@code #previous()} does not throw a {@code NoSuchElementException}.
+     */
+    public boolean hasPrevious() {
+        int previousIndex = index - 1;
+        return isWithinBounds(previousIndex);
+    }
+
+    /**
+     * Returns true if calling {@code #current()} does not throw a {@code NoSuchElementException}.
+     */
+    public boolean hasCurrent() {
+        return isWithinBounds(index);
+    }
+
+    private boolean isWithinBounds(int index) {
+        return index >= 0 && index < list.size();
+    }
+
+    /**
+     * Returns the next element in the list and advances the cursor position.
+     * @throws NoSuchElementException if there is no more next element in the list.
+     */
+    public String next() {
+        if (!hasNext()) {
+            throw new NoSuchElementException();
+        }
+        return list.get(++index);
```

Comment:
Doesn't seem like good practice, though since it's from AB4 I guess it should be fine